src/nodes/NodeGraph.ts

```diff
@@ -151,6 +151,15 @@ class NodeGraph {
     return space;
   }
 
+  @ready(new nodesErrors.ErrorNodeGraphNotRunning())
+  public async lockBucket(bucketIndex: number, tran: DBTransaction) {
```

Comment:
Should this be `protected`?

Reply:
It's getting used inside of `NodeManager` to lock the bucket during `setNode` and `garbageCollectBucket`.